include/sound/sof/header.h

```diff
@@ -108,53 +108,34 @@
 #define SOF_IPC_MSG_MAX_SIZE			384
 
 /*
- * SOF panic codes
+ * Structure Header - Header for all IPC structures except command structs.
+ * The size can be greater than the structure size and that means there is
+ * extended bespoke data beyond the end of the structure including variable
+ * arrays.
  */
-#define SOF_IPC_PANIC_MAGIC			0x0dead000
-#define SOF_IPC_PANIC_MAGIC_MASK		0x0ffff000
-#define SOF_IPC_PANIC_CODE_MASK			0x00000fff
-#define SOF_IPC_PANIC_MEM			(SOF_IPC_PANIC_MAGIC | 0x0)
-#define SOF_IPC_PANIC_WORK			(SOF_IPC_PANIC_MAGIC | 0x1)
-#define SOF_IPC_PANIC_IPC			(SOF_IPC_PANIC_MAGIC | 0x2)
-#define SOF_IPC_PANIC_ARCH			(SOF_IPC_PANIC_MAGIC | 0x3)
-#define SOF_IPC_PANIC_PLATFORM			(SOF_IPC_PANIC_MAGIC | 0x4)
-#define SOF_IPC_PANIC_TASK			(SOF_IPC_PANIC_MAGIC | 0x5)
-#define SOF_IPC_PANIC_EXCEPTION			(SOF_IPC_PANIC_MAGIC | 0x6)
-#define SOF_IPC_PANIC_DEADLOCK			(SOF_IPC_PANIC_MAGIC | 0x7)
-#define SOF_IPC_PANIC_STACK			(SOF_IPC_PANIC_MAGIC | 0x8)
-#define SOF_IPC_PANIC_IDLE			(SOF_IPC_PANIC_MAGIC | 0x9)
-#define SOF_IPC_PANIC_WFI			(SOF_IPC_PANIC_MAGIC | 0xa)
 
-/*
- * SOF memory capabilities, add new ones at the end
- */
-#define SOF_MEM_CAPS_RAM			(1 << 0)
-#define SOF_MEM_CAPS_ROM			(1 << 1)
-#define SOF_MEM_CAPS_EXT			(1 << 2) /**< external */
-#define SOF_MEM_CAPS_LP			(1 << 3) /**< low power */
-#define SOF_MEM_CAPS_HP			(1 << 4) /**< high performance */
-#define SOF_MEM_CAPS_DMA			(1 << 5) /**< DMA'able */
-#define SOF_MEM_CAPS_CACHE			(1 << 6) /**< cacheable */
-#define SOF_MEM_CAPS_EXEC			(1 << 7) /**< executable */
+struct sof_ipc_hdr {
+	uint32_t size;			/**< size of structure */
+} __packed;
```

Comment:
I find this PR difficult to double-check.
You have sof_ipc_hdr and sof_ipc_cmd_hdr which are not the same thing, but in a couple of places it's not clear which one is really expected to be used.
sof_ipc_hdr has also really nothing to do with IPC, it's just the struct size so you may want to simplify? Or are you planning to expand this sof_ipc_hdr in future revisions?

Reply:
if we use a structure then we could also add a reserved field here to be used for extensions or changes in structure layout.

Reply:
Not planning on extending, since it was meant to be a lightweight way to define size (without actually using the size keyword. I think we are good for expanding the parent structures now though.